lib/api/objectCopy.js

```diff
@@ -325,7 +330,13 @@ function objectCopy(authInfo, request, sourceBucket,
             const isVersionedObj = vcfg && vcfg.Status === 'Enabled';
             const destLocationConstraintName =
                 storeMetadataParams.dataStoreName;
-
+            if ((!serverSideEncryption && sseHeader) ||
```

Comment:
You might want to add a utility function to `isValidSSE()` lib/api/apiUtils for clarity and to avoid code duplication.